source/Irrlicht/CIrrDeviceSDL.cpp

```diff
@@ -843,13 +839,14 @@ bool CIrrDeviceSDL::activateJoysticks(core::array<SJoystickInfo> & joystickInfo)
 	int joystick = 0;
 	for (; joystick<numJoysticks; ++joystick)
 	{
-		Joysticks.push_back(SDL_JoystickOpen(joystick));
+		SDL_Joystick *opened_joystick = SDL_JoystickOpen(joystick);
+		Joysticks.push_back(opened_joystick);
 		SJoystickInfo info;
 
 		info.Joystick = joystick;
 		info.Axes = SDL_JoystickNumAxes(Joysticks[joystick]);
 		info.Buttons = SDL_JoystickNumButtons(Joysticks[joystick]);
-		info.Name = SDL_JoystickName(joystick);
+		info.Name = SDL_JoystickName(opened_joystick);

```

Comment:
This should be consistent with the other lines here, they should all use either `opened_joystick` or `Joysticks[joystick]`.